src/goto-programs/remove_complex.cpp

```diff
@@ -153,62 +155,66 @@ static void remove_complex(exprt &expr)
     }
     else if(expr.id()==ID_unary_minus)
     {
-      assert(expr.operands().size()==1);
+      auto const &unary_minus_expr = to_unary_minus_expr(expr);
       // do component-wise:
       // -x -> complex(-x.r,-x.i)
-      struct_exprt struct_expr(expr.type());
+      struct_exprt struct_expr(unary_minus_expr.type());
       struct_expr.operands().resize(2);
 
-      struct_expr.op0()=
-        unary_minus_exprt(complex_member(expr.op0(), ID_real));
+      struct_expr.op0() =
+        unary_minus_exprt(complex_member(unary_minus_expr.op(), ID_real));
 
-      struct_expr.op0().add_source_location()=expr.source_location();
+      struct_expr.op0().add_source_location() =
+        unary_minus_expr.source_location();
 
-      struct_expr.op1()=
-        unary_minus_exprt(complex_member(expr.op0(), ID_imag));
+      struct_expr.op1() =
+        unary_minus_exprt(complex_member(unary_minus_expr.op(), ID_imag));
 
-      struct_expr.op1().add_source_location()=expr.source_location();
+      struct_expr.op1().add_source_location() =
+        unary_minus_expr.source_location();
 
       expr=struct_expr;
     }
     else if(expr.id()==ID_complex)
     {
-      assert(expr.operands().size()==2);
+      static_cast<void>(to_complex_expr(expr));
       expr.id(ID_struct);
```

Comment:
Maybe we should just make this code a bit more explicit:
```
struct_exprt s(expr.type());
s.add_source_location() = expr.source_location();
s.copy_to_operands(to_complex_expr(expr).real(), to_complex_expr(expr).imag());
expr.swap(s);
```

Reply:
I've been meaning to ask, why do we prefer swap over assignment in these cases?